src/boot/efi/pe.c

```diff
@@ -262,23 +266,8 @@ static void locate_sections(
                 desc->size = section_hdr->VirtualSize;
         }
 
-        /* Sort sections
-         * NOTE: maybe bubblesort isn't ideal, but surely there aren't that many sections?
-         */
-        bool dirty;
-        do {
-                dirty = false;
-                for (size_t i = 1; i < *ret_n_sections; ++i) {
-                        int ret = strcmp8((*ret_sections)[i-1].name,
-                                          (*ret_sections)[i].name);
-                        if (ret > 0) {
-                                dirty = true;
-                                struct PeSectionDescriptor tmp = (*ret_sections)[i-1];
-                                (*ret_sections)[i-1] = (*ret_sections)[i];
-                                (*ret_sections)[i] = tmp;
-                        }
-                }
-        } while (dirty);
```

Comment:
please do not undo work earlier commits in the same PR did. we want "perfect PRs" where each commit is a logic step and there's no "undo" in them.

or in other words: do the sort_array() rework first, and then do the pe section stuff on top of it.